docs/deep-dives/data/preenrollment_bias.md

```diff
@@ -0,0 +1,116 @@
+
+Using this framework, it's quite simple to extend our experiment analysis to account for pre-experiment data. We can simply include it as a covariate in the model. We can instead estimate:
+
+$$y_i = \beta_0 + \beta_t t_i + \beta_y z_i$$
+
+Where $z_i$ is the metric of interest ($y$) for the $i$-th unit as measured during the pre-experiment period. As before, we're interested in inferences to $\beta_t$, but now these inferences will be:
+
+1. Adjusted to account for pre-experiment information
+2. Benefit from an increase in precision.
+
+### Configuring covariate adjustment
+
+#### Adjusting a new metric for preenrollment bias
+
+To perform adjustment for a new metric, you can write/edit the [custom config](../jetstream/configuration.md#custom-experiment-configurations) to do 2 things: 1) configure your metric to be calculated over the preenrollment period (that is, perform the retrospective A/A test) and 2) configure the adjustment.
+
+To ensure that your metric is computed during the pre-enrollment period, simply add it to the desired period metric list:
+
+```toml
+preenrollment_weekly = [
+    'my_new_metric'
+]
+```
+
+To configure the adjustment, first designate that inferences on the mean are desired using linear models. Then, configure that statistic to adjust based on the period chosen above. For example:
+
+```toml
+[metrics.my_new_metric.statistics.linear_model_mean] # desire to estimate the mean using linear models
+[metrics.my_new_metric.statistics.linear_model_mean.covariate_adjustment] # desire to adjust that estimate
+period = "preenrollment_week" # adjust using the same metric calculated during the week prior to enrollment
+```
+
+One can reference how adjustment is configured for guardrails ([example](https://github.com/mozilla/metric-hub/blob/main/jetstream/defaults/firefox_desktop.toml#L33-L67)).
+
+:::note
+Currently, the custom configs only support adjusting a during-treatment metric using the pre-experiment version of that metric. It's not supported to adjust a metric using a different metric or by using during-experiment data. To accomplish either of those tasks, you'll need to do so manually.
+:::
+
+:::info
+As of February 2025, the execution order of analysis periods is not guaranteed. This means that, when rerunning an analysis for an experiment, it's possible for the computation for the during-treatment analysis to execute before the preenrollment has finished. This will result in the adjustment not being performed. That is, Jetstream will automatically fall back to unadjusted inferences. You can determine if Jetstream fell back by either examining the logs or by comparing to the unadjusted confidence intervals (which will be identical if adjustment was not performed).
```

Comment:
```suggestion
As of February 2025, the execution order of analysis periods is not guaranteed. This means that, when rerunning an analysis for an experiment, it's possible for the computation for the during-treatment analysis to execute before the preenrollment has finished. This will result in the adjustment not being performed. That is, Jetstream will automatically fall back to unadjusted inferences. You can determine if Jetstream fell back by either examining the logs (see [dashboard](https://mozilla.cloud.looker.com/dashboards/246?Experiment=&Timestamp+Date=14+day&Log+Level=ERROR%2CWARNING)) or by comparing to the unadjusted confidence intervals (which will be identical if adjustment was not performed).
```